usockets and uwebsockets needs update to latest

uwebsockets needs to be updated to the latest release. The build is failing:
```
external/uwebsockets/src/App.h:90:62: error: static assertion failed: Mismatching uSockets/uWebSockets ABI
   90 |     static_assert(sizeof(struct us_socket_context_options_t) == sizeof(SocketContextOptions), "Mismatching uSockets/uWebSockets ABI");
```

It looks like the latest µWebSockets may depend on an unreleased µSockets? (The latest µSockets release is Sept 2021. There should perhaps also be a note that these packages should update in tandem in the relevant `repository.bzl`(s)?)

Note also that `new_release.py` is not currently working on usockets due to sigmavirus24/github3.py#1105.

--- a/tools/workspace/usockets/repository.bzl
+++ b/tools/workspace/usockets/repository.bzl
@@ -8,8 +8,10 @@ def usockets_repository(
     github_archive(
         name = name,
         repository = "uNetworking/uSockets",
-        commit = "v0.8.1",
-        sha256 = "3b33b5924a92577854e2326b3e2d393849ec00beb865a1271bf24c0f210cc1d6",  # noqa
+        # This should match the SHA referenced by uwebsockets as a git
+        # submodule.
+        commit = "9d3d53af5658f7a472c4fd2c0c0a837c0baa93f0",
+        sha256 = "562ee841afa741144f49a982ed7ebc298928918f29c2e35957eba4896f651dc8",  # noqa
         build_file = ":package.BUILD.bazel",
         mirrors = mirrors,
     )

--- a/tools/workspace/uwebsockets/repository.bzl
+++ b/tools/workspace/uwebsockets/repository.bzl
@@ -8,8 +8,8 @@ def uwebsockets_repository(
     github_archive(
         name = name,
         repository = "uNetworking/uWebSockets",
-        commit = "v20.14.0",
-        sha256 = "15cf995844a930c9a36747e8d714b94ff886b6814b5d4e3b3ee176f05681cccc",  # noqa
+        commit = "v20.16.0",
+        sha256 = "8847cd9f238cc71e1b0ad52beec7c55c5affc5fad4bb32c3840832a2621ecca3",  # noqa
         build_file = ":package.BUILD.bazel",
         mirrors = mirrors,
     )

